phase4-lib/src/main/java/com/helger/phase4/servlet/AS4IncomingHandler.java

```diff
@@ -953,7 +954,10 @@ public static Ebms3UserMessage parseUserMessage (@Nonnull final IAS4CryptoFactor
                                                    @Nonnull final byte [] aResponsePayload,
                                                    @Nullable final IAS4IncomingDumper aIncomingDumper,
                                                    @Nonnull final IAS4IncomingSecurityConfiguration aIncomingSecurityConfiguration,
-                                                   @Nullable final IAS4UserMessageConsumer aUserMsgConsumer) throws Phase4Exception
+                                                   @Nullable final IAS4UserMessageConsumer aUserMsgConsumer,

```

Comment:
I am not really in favour of this API extension, as the message is explicitly called `parseUserMessage`.  The counterpart method is `parseSignalMessage`.

Reply:
You are right, I will move this one level higher and use the parseSignalMessage if no UserMessage is present